src/neuroconv/tools/testing/mock_ttl_signals.py

```diff
@@ -0,0 +1,290 @@
+        signal_mean_int16_default += shift
+
+    if np.issubdtype(dtype, np.integer):
+        baseline_mean = baseline_mean or baseline_mean_int16_default
+        signal_mean = signal_mean or signal_mean_int16_default
+        channel_noise = channel_noise or channel_noise_int16_default
+        generic_dtype = np.integer
+    else:
+        baseline_mean = baseline_mean or baseline_mean_int16_default * default_gain_to_volts
+        signal_mean = signal_mean or signal_mean_int16_default * default_gain_to_volts
+        channel_noise = channel_noise or channel_noise_int16_default * default_gain_to_volts
+        generic_dtype = np.floating
+    parameters_to_check = dict(baseline_mean=baseline_mean, signal_mean=signal_mean, channel_noise=channel_noise)
+    for parameter_name, parameter_value in parameters_to_check.items():
+        _check_parameter_dtype_consistency(
+            parameter_name=parameter_name, parameter_value=parameter_value, generic_dtype=generic_dtype
+        )
+
+    np.random.seed(seed=random_seed)
+    num_frames = np.ceil(signal_duration * sampling_frequency_hz).astype(int)
+    trace = (np.random.randn(num_frames) * channel_noise + baseline_mean).astype(dtype)
+
+    if ttl_times is not None:
+        ttl_times = np.array(ttl_times)
+    else:
+        ttl_times = np.arange(start=1.0, stop=signal_duration, step=2.0)
+    assert not any(
+        np.diff(ttl_times) <= ttl_duration
+    ), "There are overlapping TTL 'on' intervals! Please specify disjoint on/off periods."
+
+    ttl_start_frames = np.round(ttl_times * sampling_frequency_hz).astype(int)
+    num_frames_ttl_duration = np.round(ttl_duration * sampling_frequency_hz).astype(int)
+    ttl_intervals = (slice(start, start + num_frames_ttl_duration) for start in ttl_start_frames)
+
+    for ttl_interval in ttl_intervals:
+        trace[ttl_interval] += signal_mean
+
+    return trace
+
+
+def regenerate_test_cases(folder_path: FolderPathType, regenerate_reference_images: bool = False):  # pragma: no cover
+    """
+    Regenerate the test cases of the file included in the main testing suite, which is frozen between breaking changes.
+
+    Parameters
+    ----------
+    folder_path: PathType
+        Folder to save the resulting NWB file in. For use in the testing suite, this must be the
+        '/test_testing/test_mock_ttl/' subfolder adjacent to the 'test_mock_tt.py' file.
+    regenerate_reference_images: boolean
+        If true, uses the kaleido package with plotly (you may need to install both) to regenerate the images used
+        as references in the documentation.
+    """
+    folder_path = Path(folder_path)
+
+    if regenerate_reference_images:
+        assert is_module_installed("plotly") and is_module_installed("kaleido"), (
+            "To regenerate the reference images, " "you must install both plotly and kaleido!"
+        )
+        import plotly.graph_objects as go
+        from plotly.subplots import make_subplots
```

Comment:
Seems like a strange choice to use plotly and kaleido here if you want to end up with a .png. I feel like matplotlib would have been much more natural.

Reply:
Personal preference - I prefer `plotly` as a default (and you don't usually need `kaleido` to use the 'save as' button in Jupyter - that's just for doing it purely in code)

This function only needs to be called when you alter source code of the generation function, so possibly never again. Really worth the time/effort to refactor to matplotlib?

Reply:
It's fine for now but make sure you have these listed as deps on the appropriate place